dhall/src/Dhall/Syntax.hs

```diff
@@ -289,33 +291,70 @@ instance Bifunctor PreferAnnotation where
 
 For example,
 
-> { {- A -} x : T }
+> { {- A -} x {- B -} : {- C -} T }
 
 ... or
 
-> { {- A -} x = T }
+> { {- A -} x {- B -} = {- C -} T }
 
 will be instantiated as follows:
 
-* @recordFieldSrc@ corresponds to the @A@ comment.
-* @field@ is @"T"@
+* @recordFieldSrc0@ corresponds to the @A@ comment.
+* @recordFieldValue@ is @"T"@
+* @recordFieldSrc1@ corresponds to the @B@ comment.
+* @recordFieldSrc2@ corresponds to the @C@ comment.
 
 Although the @A@ comment isn't annotating the @"T"@ Record Field,
-this is the best place to keep these comments
+this is the best place to keep these comments.
+
+Note that @recordFieldSrc2@ is always 'Nothing' when the 'RecordField' is for
+a punned entry, because there is no @=@ sign. For example,
+
+> { {- A -} x {- B -} }
+
+will be instantiated as follows:
+
+* @recordFieldSrc0@ corresponds to the @A@ comment.
+* @recordFieldValue@ corresponds to @(Var "x")@
+* @recordFieldSrc1@ corresponds to the @B@ comment.
+* @recordFieldSrc2@ will be 'Nothing'
+
+If dot-syntax is being used all three srcs are 'Just's, but for
+every label except the last one @recordFieldSrc1@ should be equal to
+@recordFieldSrc2@. On the last label, @recordFieldSrc2@ semantics are the same
```

Comment:
Why do we have this duplication? Why can't `recordFieldSrc2` simply be `Nothing` when the label isn't directly followed by a `=`?

Reply:
You're right. My rationale was to have a way to differ dot-syntax from punned entries, but I didn't notice that by doing this we don't have a way to differ between dot-syntax and normal labels

Reply:
I changed the semantics as you suggested, @sjakobi!